tajo-core/src/main/java/org/apache/tajo/ha/HdfsServiceTracker.java

```diff
@@ -70,12 +70,45 @@
 
   public HdfsServiceTracker(TajoConf conf) throws IOException {
     this.conf = conf;
+  }
+
+  @Override
+  public void register() throws IOException {
     initSystemDirectory();
 
     InetSocketAddress socketAddress = conf.getSocketAddrVar(ConfVars.TAJO_MASTER_UMBILICAL_RPC_ADDRESS);
-    this.masterName = socketAddress.getAddress().getHostAddress() + ":" + socketAddress.getPort();
+    this.masterName = socketAddress.getAddress().getHostName()+ ":" + socketAddress.getPort();
```

Comment:
If a hostname is not FQDN, it may be throw the unknown host exception.
In my opinion, ip address is more safety. How do you think?